config/src/main/java/org/axonframework/config/AggregateConfigurer.java

```diff
@@ -249,12 +250,22 @@ protected AggregateConfigurer(Class<A> aggregate) {
                     }
                     return builder.build();
                 });
+        commandHandlerAggregateFactory = new Component<Function<Object, A>>(
```

Comment:
Could we make this an interface instead of a Function? This way it can also be configured in Spring using `@Bean`

Reply:
I don't think that actually makes a difference. A function (Functional Interface) is simply an interface with a single abstract method, which allows the shortcut notation to be hsed to declare it. The annotation is just a hint for the compiler to validate that there is only one abstract method. 
You can simply define a Spring bean of the Functional Interface type and return a lambda as its implementation.